### PR TITLE
[Arch] Removing docker exec box

### PR DESCRIPTION
**Pull request by xingyaoww**
**What is the problem that this fixes or functionality that this introduces? Does it fix any open issues?**

`DockerExecBox` does not support stateful commands, which makes it not so useful at the current state of OpenDevin. We've been talking about deprecating it for a while - i think it is time to start *deleting* code instead of adding more code to make the upcoming architecture refractor easier: https://github.com/OpenDevin/OpenDevin/issues/2404

**Give a brief summary of what the PR does, explaining any non-trivial design decisions**

Remove `DockerExecBox`, and related documentations and workflows.

**Other references**

### CODE REVIEW

**Review by tobitege — APPROVED** [review]
Looks good to me, afaict.

Good riddance, execbox 😬

File: tests/unit/test_sandbox.py [hunk 1]
```diff
@@ -137,7 +136,7 @@ def test_ssh_box_multi_line_cmd_run_as_devin(temp_dir):
     ), patch.object(config, 'run_as_devin', new='true'), patch.object(
         config, 'sandbox_type', new='ssh'
     ):
-        for box in [DockerSSHBox(), DockerExecBox()]:
+        for box in [DockerSSHBox()]:
```
Comment (enyst) [inline tests/unit/test_sandbox.py:139]:
```suggestion
        box = DockerSSHBox()
```

File: tests/unit/test_sandbox.py [hunk 2]
```diff
@@ -158,9 +157,7 @@ def test_ssh_box_stateful_cmd_run_as_devin(temp_dir):
     ), patch.object(config, 'run_as_devin', new='true'), patch.object(
         config, 'sandbox_type', new='ssh'
     ):
-        for box in [
-            DockerSSHBox()
-        ]:  # FIXME: DockerExecBox() does not work with stateful commands
+        for box in [DockerSSHBox()]:
```
Comment (enyst) [inline tests/unit/test_sandbox.py:160]: Same, we can simplify a bit this code that used to expect multiple boxes when it doesn't anymore 😅

Comment (xingyaoww) [reply tests/unit/test_sandbox.py]: done! replaced all occurrences of `[DockerSSHBox()]` with assignment!

**Review by li-boxuan — APPROVED** [review]

**Pull request merged**